platform/darwin/src/MGLNetworkConfiguration.m

```diff
@@ -48,4 +62,53 @@ - (NSURLSessionConfiguration *)defaultSessionConfiguration {
     return sessionConfiguration;
 }
 
+- (void)startDownloadEvent:(NSString *)urlString type:(NSString *)resourceType {
+    if (urlString && ![self.events objectForKey:urlString]) {
+        [self.events setObject:@{ MGLStartTime: [NSDate date], MGLResourceType: resourceType } forKey:urlString];
+    }
+}
+
+- (void)stopDownloadEvent:(NSString *)urlString {
+    [self sendEventForURL:urlString withAction:nil];
+}
+
+- (void)cancelDownloadEvent:(NSString *)urlString {
+    [self sendEventForURL:urlString withAction:@"cancel"];
+}
+
+- (void)sendEventForURL:(NSString *)urlString withAction:(NSString *)action
+{
+    if (urlString && [self.events objectForKey:urlString]) {
+        NSDictionary *eventAttributes = [self eventAttributesForURL:urlString withAction:action];
+        [self.metricsDelegate networkConfiguration:self didGenerateMetricEvent:eventAttributes];
+        [self.events removeObjectForKey:urlString];
+    }
+}
+
+- (NSDictionary *)eventAttributesForURL:(NSString *)urlString withAction:(NSString *)action
+{
+    NSDictionary *parameters = [self.events objectForKey:urlString];
+    NSDate *startDate = [parameters objectForKey:MGLStartTime];
+    NSDate *endDate = [NSDate date];
+    NSTimeInterval elapsedTime = [endDate timeIntervalSinceDate:startDate];
+    NSDateFormatter* iso8601Formatter = [[NSDateFormatter alloc] init];
```

Comment:
For later: we should look at supporting `NSISO8601DateFormatter` where available.